cmd/keymasterd/main.go

```diff
@@ -217,8 +283,10 @@ func checkUserPassword(username string, password string, config AppConfigFile) (
 		if err != nil {
 			return false, err
 		}
+		go metricLogAuthOperation(clientType, "password", valid)
 		return valid, nil
 	}
+	go metricLogAuthOperation(clientType, "password", false)
```

Comment:
Why do this in a goroutine? These operations should be very fast.
Same comment applies elsewhere.

Reply:
because they get hold of locks... I rather push them to a separate routine to ensure that the main operation is not blocked unnecesarily.